### PR TITLE
Switch from working with "Team: Foo" labels to "Product Area: Bar"

### PR DESCRIPTION
**Pull request by chadwhitacre**
This is for https://github.com/getsentry/team-ospo/issues/126 which is for https://github.com/getsentry/team-ospo/issues/90.

It's a lot. I'm not going to say it's not a lot.

### CODE REVIEW

File: src/buildServer.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@
   server.post(
     '/webhooks/pubsub',
     PubSub.opts,
-    PubSub.notifyTeamsForUntriagedIssues
+    PubSub.notifyProductOwnersForUntriagedIssues
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/buildServer.ts:105]: Not fixing here since I didn't introduce it here, reticketed as https://github.com/getsentry/eng-pipes/issues/437.

**Comment by hubertdeng123** [comment]
Looks like this PR will break the hourly triage notifications that exist today, since `Team: *` labels still exist. With that said, is there a plan to clean the existing database so that the `Team: *` labels no longer exist as entries in `label_to_channel`? Will channels that Product Owners subscribe to be created before the team labels are switched to product area labels on GH issues?

File: src/utils/businessHours.ts [hunk 2]
```diff
@@ -147,14 +151,14 @@ export const isChannelInBusinessHours = async (
...
 export async function getNextAvailableBusinessHourWindow(
-  team,
+  productArea,
   momentTime
 ): Promise<BusinessHourWindow> {
-  let offices = await getOffices(team);
+  let offices = await getOffices(productArea);
   if (offices.length === 0) {
-    offices = await getOffices('Team: Open Source');
+    offices = await getOffices('Product Area: Open Source');
```
Comment (hubertdeng123) [inline src/utils/businessHours.ts:159]: This product area won't exist, right? Do we need to hard code offices for this case instead?

File: src/webhooks/pubsub/index.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ import { db } from '@utils/db';
 
 const DEFAULT_REPOS = [SENTRY_REPO];
 const GH_API_PER_PAGE = 100;
-const DEFAULT_TEAM_LABEL = 'Team: Open Source';
+const DEFAULT_PRODUCT_AREA_LABEL = 'Product Area: Open Source';
```
Comment (hubertdeng123) [inline src/webhooks/pubsub/index.ts:22]: Should this still be the default product area label since `Product Area: Open Source` doesn't exist?

**Review by hubertdeng123 — CHANGES_REQUESTED** [review]

File: src/webhooks/pubsub/index.ts (hunk 3 above)
Comment (chadwhitacre) [reply src/webhooks/pubsub/index.ts]: Good catch, done in 934ea3b57cf731ba7259b72bdcc2b300f96c7fb4.

File: src/brain/issueLabelHandler/route.ts [hunk 4]
```diff
@@ -103,7 +104,7 @@ export async function markUnrouted({
 
   const timeToRouteBy = await calculateSLOViolationRoute(UNROUTED_LABEL);
   const { readableDueByDate, lastOfficeInBusinessHours } =
-    await getReadableTimeStamp(timeToRouteBy, 'Team: Support');
+    await getReadableTimeStamp(timeToRouteBy, 'Product Area: Support');
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/brain/issueLabelHandler/route.ts]: Done in 5d7bfba2c1e5f9d50e57ebc4499aaf03817b43a2. 👍

Will need to add this as a label in the repo ...

File: src/utils/businessHours.ts (hunk 2 above)
Comment (hubertdeng123) [reply src/utils/businessHours.ts]: Yep we can, technically I think our team or Support should handle all cases when Product Area doesn't have a clear product owner

**Comment by hubertdeng123** [comment]
`Find all Slack "Team: Foo" subscriptions and set up corresponding "Product Area: Foo" subscriptions` might be a bit difficult without access to the database 🤔. Probably could ask ops to run a query for this?

**Review by hubertdeng123 — DISMISSED** [review]

**Review by hubertdeng123 — APPROVED** [review]
just tested the migration locally, seems to be working as expected
Steps:
1. On main, subscribe to notifications for Open Source in yyz in a slack channel #random
2. Ensure the channel #random has been subscribed to Open Source
3. Check out branch, run migration
4. In the #random channel, check to make sure there are no slack subscriptions

File: src/brain/issueLabelHandler/route.ts [hunk 5]
```diff
@@ -114,10 +115,17 @@ export async function markUnrouted({
...
+  productAreaLabelName,
+  productAreaLabelDescription
+) {
   try {
     const strippedTeamName =
-      teamLabelName?.substr(6).replace(' ', '-').toLowerCase() || '';
+      productAreaLabelName
+        ?.substr(LENGTH_OF_PRODUCT_AREA_LABEL_PREFIX)
+        .replace(' ', '-')
+        .toLowerCase() || '';
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/brain/issueLabelHandler/route.ts:128]: Took care of this in #439.

File: src/brain/issueLabelHandler/route.ts [hunk 6]
```diff
@@ -65,7 +66,7 @@ function isValidLabel(payload) {
 
 function shouldLabelBeRemoved(label, target_name) {
   return (
-    (label.name.startsWith('Team: ') && label.name !== target_name) ||
+    (label.name.startsWith('Product Area: ') && label.name !== target_name) ||
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/brain/issueLabelHandler/route.ts:69]: #445! :)

File: src/brain/issueLabelHandler/route.ts [hunk 7]
```diff
@@ -17,6 +17,7 @@ import {
 import { getOssUserType } from '@utils/getOssUserType';
 import { isFromABot } from '@utils/isFromABot';
 
+const LENGTH_OF_PRODUCT_AREA_LABEL_PREFIX = 14;
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/brain/issueLabelHandler/route.ts:20]: #445! :)

File: src/brain/issueLabelHandler/route.ts [hunk 8]
```diff
@@ -103,7 +104,7 @@ export async function markUnrouted({
 
   const timeToRouteBy = await calculateSLOViolationRoute(UNROUTED_LABEL);
   const { readableDueByDate, lastOfficeInBusinessHours } =
-    await getReadableTimeStamp(timeToRouteBy, 'Team: Support');
+    await getReadableTimeStamp(timeToRouteBy, 'Product Area: Unknown');
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/brain/issueLabelHandler/route.ts:107]: #445! :)

**Pull request merged**